worker/uniter/charm/bundles.go

```diff
@@ -103,8 +103,16 @@ func (d *BundlesDir) bundleURLPath(url *charm.URL) string {
 	return path.Join(d.path, charm.Quote(url.String()))
 }
 
+// ClearDownloads removes any entries in the temporary bundle download
+// directory. It is intended to be called on uniter startup.
+func ClearDownloads(bundlesDir string) error {
+	downloadDir := downloadsPath(bundlesDir)
+	err := os.RemoveAll(downloadDir)
+	return errors.Annotate(err, "unable to clear bundle downloads")
+}
+
 // downloadsPath returns the path to the directory into which charms are
 // downloaded.
-func (d *BundlesDir) downloadsPath() string {
-	return path.Join(d.path, "downloads")
+func downloadsPath(bunsDir string) string {
```

Comment:
To make it clear that it's supposed to receive the bundles directory. `bunsDir` is already used throughout this file so I was being consistent.